src/mongo/db/pipeline/expression_test.cpp

```diff
@@ -139,6 +144,239 @@ class ExpressionNaryTestOneArg : public ExpressionNaryTest {
+class Testable : public ExpressionNary {
+public:
+    virtual Value evaluateInternal(Variables* vars) const {
+        // Just put all the values in a list.
+        // By default, this is not associative/commutative so the results will change if
+        // instantiated as commutative or associative and operations are reordered.
+        vector<Value> values;
+        for (ExpressionVector::const_iterator i = vpOperand.begin(); i != vpOperand.end(); ++i) {
+            values.push_back((*i)->evaluateInternal(vars));
+        }
+        return Value(values);
+    }
+
+    virtual const char* getOpName() const {
+        return "$testable";
+    }
+
+    virtual bool isAssociative() const {
+        return _isAssociative;
+    }
+
+    virtual bool isCommutative() const {
+        return _isCommutative;
+    }
+
+    static intrusive_ptr<Testable> create(bool associative, bool commutative) {
+        return new Testable(associative, commutative);
+    }
+
+private:
+    Testable(bool isAssociative, bool isCommutative)
+        : _isAssociative(isAssociative), _isCommutative(isCommutative) {}
+    bool _isAssociative;
+    bool _isCommutative;
+};
+
+class ExpressionNaryTest : public unittest::Test {
+public:
+    virtual void setUp() override {
+        _notAssociativeNorCommutative = Testable::create(false, false);
+        _associativeOnly = Testable::create(true, false);
+        _associativeAndCommutative = Testable::create(true, true);
+    }
+
+protected:
+    void assertDependencies(const intrusive_ptr<Testable>& expr,
+                            const BSONArray& expectedDependencies) {
+        DepsTracker dependencies;
+        expr->addDependencies(&dependencies);
+        BSONArrayBuilder dependenciesBson;
+        for (set<string>::const_iterator i = dependencies.fields.begin();
+             i != dependencies.fields.end();
+             ++i) {
+            dependenciesBson << *i;
+        }
+        ASSERT_EQUALS(expectedDependencies, dependenciesBson.arr());
+        ASSERT_EQUALS(false, dependencies.needWholeDocument);
+        ASSERT_EQUALS(false, dependencies.needTextScore);
+    }
+
+    void assertContents(const intrusive_ptr<Testable>& expr, const BSONArray& expectedContents) {
+        ASSERT_EQUALS(constify(BSON("$testable" << expectedContents)), expressionToBson(expr));
+    }
+
+    void addOperandArrayToExpr(const intrusive_ptr<Testable>& expr, const BSONArray& operands) {
+        VariablesIdGenerator idGenerator;
+        VariablesParseState vps(&idGenerator);
+        BSONObjIterator i(operands);
+        while (i.more()) {
+            BSONElement element = i.next();
+            expr->addOperand(Expression::parseOperand(element, vps));
+        }
+    }
+
+    intrusive_ptr<Testable> _notAssociativeNorCommutative;
+    intrusive_ptr<Testable> _associativeOnly;
+    intrusive_ptr<Testable> _associativeAndCommutative;
+};
+
+TEST_F(ExpressionNaryTest, AddOperand) {
```

Comment:
Thank you for converting these to the new-style unit tests! While you're here, we're making an effort to improve the names of new tests to make it easier to understand what they're testing. For example, instead of AddOperand, maybe AddedOperandsShouldAppearInSerialization. This will make it easier to scan the file to find obvious testing gaps. If you could update the names to this new format, it would be much appreciated. When it's not possible (without an absurdly long name), consider splitting into two or more tests, or adding a descriptive comment, as some of the tests already have.